samples/apps/copilot-chat-app/webapi/SemanticKernelExtensions.cs

```diff
@@ -67,27 +67,39 @@ internal static IServiceCollection AddSemanticKernelServices(this IServiceCollec
+    private static void AddSemanticTextMemory(this IServiceCollection services)
     {
-        MemoriesStoreOptions config = serviceProvider.GetRequiredService<IOptions<MemoriesStoreOptions>>().Value;
+        MemoriesStoreOptions config = services.BuildServiceProvider().GetRequiredService<IOptions<MemoriesStoreOptions>>().Value;
         switch (config.Type)
         {
             case MemoriesStoreOptions.MemoriesStoreType.Volatile:
-                return new SemanticTextMemory(
-                    new VolatileMemoryStore(),
-                    serviceProvider.GetRequiredService<IOptionsSnapshot<AIServiceOptions>>().Get(AIServiceOptions.EmbeddingPropertyName)
-                        .ToTextEmbeddingsService(logger: serviceProvider.GetRequiredService<ILogger<AIServiceOptions>>()));
+            {
+                services.AddSingleton<IMemoryStore, VolatileMemoryStore>();
+                services.AddScoped<ISemanticTextMemory>(sp => new SemanticTextMemory(
+                    sp.GetRequiredService<IMemoryStore>(),
+                    sp.GetRequiredService<IOptionsSnapshot<AIServiceOptions>>().Get(AIServiceOptions.EmbeddingPropertyName)
+                        .ToTextEmbeddingsService(logger: sp.GetRequiredService<ILogger<AIServiceOptions>>())));
+            }
+            break;
 
             case MemoriesStoreOptions.MemoriesStoreType.Qdrant:
-                return new SemanticTextMemory(
-                    new QdrantMemoryStore(config.Qdrant!.Host, config.Qdrant.Port, config.Qdrant.VectorSize, serviceProvider.GetRequiredService<ILogger<QdrantMemoryStore>>()),
-                    serviceProvider.GetRequiredService<IOptionsSnapshot<AIServiceOptions>>().Get(AIServiceOptions.EmbeddingPropertyName)
-                        .ToTextEmbeddingsService(logger: serviceProvider.GetRequiredService<ILogger<AIServiceOptions>>()));
+            {
+                services.AddSingleton<IMemoryStore>(sp => new QdrantMemoryStore(
+                    config.Qdrant!.Host, config.Qdrant.Port, config.Qdrant.VectorSize, sp.GetRequiredService<ILogger<QdrantMemoryStore>>()));
+                services.AddScoped<ISemanticTextMemory>(sp => new SemanticTextMemory(
+                    sp.GetRequiredService<IMemoryStore>(),
+                    sp.GetRequiredService<IOptionsSnapshot<AIServiceOptions>>().Get(AIServiceOptions.EmbeddingPropertyName)
+                        .ToTextEmbeddingsService(logger: sp.GetRequiredService<ILogger<AIServiceOptions>>())));
+            }
+            break;
 
             case MemoriesStoreOptions.MemoriesStoreType.AzureCognitiveSearch:
-                return new AzureCognitiveSearchMemory(config.AzureCognitiveSearch!.Endpoint, config.AzureCognitiveSearch.Key);
+            {
+                services.AddSingleton<ISemanticTextMemory>(sp => new AzureCognitiveSearchMemory(config.AzureCognitiveSearch!.Endpoint, config.AzureCognitiveSearch.Key));
```

Comment:
How do we know for a fact that config.AzureCognitiveSearch is not null? (I'm asking because of the presence of the '!') Maybe we should check!

Reply:
It is verified in curing Options validation. I'll put checks in anyway - doesn't hurt.